Alex and Happy Part 1/article.md

```diff
@@ -365,16 +381,20 @@ import Data.ByteString.Lazy.Char8 (ByteString)
 import qualified Data.ByteString.Lazy.Char8 as BS
 ```
 
-### Regex and regex macros
+### Lexing identifiers
 
-Alex uses regular expressions to match lexemes, so we will use them to recognize the various lexemes in our language.
 Let's start with a simple one: identifiers.
 We are free to choose the specification for identifiers, but let's use the following:
 
 > An identifier is a sequence of alphanumeric characters, primes ('), question marks (?), and underscores (_).
 > The identifier must begin with a letter or an underscore.
 
-Alex allows the creation of _character set macros_, which are shortcuts that you can use to avoid duplicating macros. We use `$NAME = CHARACTER_SET` to define a character set macro.
+<hr>
+<b>Regex macros</b>
+
+To make it easier to write the regex pattern for identifiers, we can use character set and regex macros.
```

Comment:
Sorry for bothering, but I think this sentence needs to be outside so people understand for what the box is there. 😁